discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -179,33 +188,36 @@ def multihash_is_directory(self, multihash):
                 logger.warning(f"IPFSCLIENT | Found directory {multihash}")
                 return True
 
-        # Attempt to retrieve from cnode gateway endpoints.
+        # If not found via IPFS, attempt to retrieve from cnode gateway endpoints.
         gateway_endpoints = self._cnode_endpoints
         for address in gateway_endpoints:
             # First, query as dir.
-            gateway_query_address = urljoin(address, f"/ipfs/{multihash}/150x150.jpg")
+            gateway_query_address = construct_image_dir_gateway_url(address, multihash, is_square)
```

Comment:
so i think this is an improvement for sure since we check `640x.jpg` instaed of always `150x150.jpg`, is there potentially value in falling back to `original.jpg` for both? not blocking

Reply:
i think original.jpg for both makes the most sense

Reply:
fwiw the reason for not using original.jpg by default is bc the creator node will be returning the _entire_ response even if we don't need it - by using the smallest possible size we can avoid serving extra bytes. however original.jpg is the only shared size between the two

Reply:
are we streaming this from our own /ipfs/Qm.../ gateways? if so we could just specificy a content range header and request 1 byte. OR we could just send a HEAD request which fetches none of the content

Reply:
this only gets requested to the gateways if IPFS is not able to find the content. 

a HEAD request seems like the right answer here. we want to know if the request will pass without the body so that's exactly what HEAD is useful for. also i'm not too worried about this interaction creating a lot of load on the cnode gateways. this route only runs if IPFS fails to fetch the content and only runs once per CID per discovery service during indexing. compared to the times actual image uufiles are requested client side, it's basically negligible.